aws/resource_aws_glue_catalog_crawler.go

```diff
@@ -0,0 +1,411 @@
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/structure"
+)
+
+func resourceAwsGlueCatalogCrawler() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueCatalogCrawlerCreate,
+		Read:   resourceAwsGlueCatalogCrawlerRead,
+		Update: resourceAwsGlueCatalogCrawlerUpdate,
+		Delete: resourceAwsGlueCatalogCrawlerDelete,
+		Exists: resourceAwsGlueCatalogCrawlerExists,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"database_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"role": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"schedule": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"classifiers": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"schema_change_policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delete_behavior": {
+							Type:     schema.TypeString,
+							Optional: true,
+							//ValidateFunc: validateDeletion,
+							//TODO: Write a validate function to ensure value matches enum
+						},
+						"update_behavior": {
+							Type:     schema.TypeString,
+							Optional: true,
+							//ValidateFunc: validateUpdate,
+							//TODO: Write a validate function to ensure value matches enum
+						},
+					},
+				},
+			},
+			"table_prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"s3_target": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"exclusions": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"jdbc_target": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"connection_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"path": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"exclusions": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+			"configuration": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
```

Comment:
We should prefer to use the generic JSON DiffSuppressFunc, `suppressEquivalentJsonDiffs`, here as this does not look like an IAM policy according to the SDK documentation:

```
// Example: '{ "Version": 1.0, "CrawlerOutput": { "Partitions": { "AddOrUpdateBehavior":
// "InheritFromTable" } } }'
```

Reply:
Done

Reply:
Done